packages/form-builder/src/index.tsx

```diff
@@ -1,45 +1,24 @@
 import React from 'react';
 import {createRoot} from 'react-dom/client';
-import {BlockSupports, registerBlockType} from '@wordpress/blocks';
-
-import App from './App';
 
 import sectionBlocks, {sectionBlockNames} from './blocks/section';
 import fieldBlocks from './blocks/fields';
 import elementBlocks from './blocks/elements';
-import {FieldBlock} from '@givewp/form-builder/types';
-
-const supportOverrides: BlockSupports = {
-    customClassName: false,
-    html: false,
-};
+import extensionBlocks from './blocks/extensions';
+import blockRegistrar from '@givewp/form-builder/common/registrars/blocks';
 
-sectionBlocks.map(({name, settings}: FieldBlock) =>
-    registerBlockType(name, {
-        ...settings,
-        supports: {
-            ...settings.supports,
-            ...supportOverrides,
-        },
-    })
-);
+import App from './App';
 
-[...fieldBlocks, ...elementBlocks].map(({name, settings}: FieldBlock) =>
-    registerBlockType(name, {
-        ...settings,
-        parent: sectionBlockNames,
-        supports: {
-            ...settings.supports,
-            ...supportOverrides,
-        },
-    })
-);
+blockRegistrar.register(sectionBlocks);
+blockRegistrar.register(fieldBlocks, sectionBlockNames);
+blockRegistrar.register(elementBlocks, sectionBlockNames);
+blockRegistrar.register(extensionBlocks, sectionBlockNames);
```

Comment:
@pauloiankoski this is looking great.  I think we're going to want to elevate the `blockRegistrar` to the window and have it load as a standalone script _before_ the form builder and add-ons.  

This would mean both add-ons and the form-builder would do this instead:

`window.givewp.blocks.registrar.register()`

I would even consider registering our core blocks before add-ons to prevent any conflicts.  That could be a nice seperation between our blocks and the form builder instance itself - which is how it's done in wp core.